### PR TITLE
OCPBUGS-34050: Backport 125337

### PR DESCRIPTION
**Pull request by JoelSpeed**
This is picking down a change to the way node IPs are set when cloud provider external is being used. In particular, the node will pick up the IP for the default route for the correct family.

This should allow us to recover from hibernation states where the Kube API server is unable to serve a Node with correctly set CCM addresses.

### CODE REVIEW

**Comment by dinhxuanvu** [comment]
I think we can merge this PR without impacting the current 1.30 rebase assuming it passes CI and blocker payload test. I will lgtm on this PR when it passes those tests.
/hold cancel

**Review by dinhxuanvu — APPROVED** [review]
/lgtm

**Comment by dinhxuanvu** [comment]
/hold
Hold for the payload test. lgtm by accident :(

**Comment by dinhxuanvu** [comment]
/payload periodic-ci-openshift-release-master-nightly-4.17-e2e-metal-ipi-ovn-bm

**Comment by JoelSpeed** [comment]
/hold cancel

Tests are looking good. Metal failure in the payload job is a known issue.

**Pull request merged**
